src/jit/regset.cpp

```diff
@@ -3570,49 +3570,38 @@ bool genIsProperRegPair(regPairNo regPair)
 
 regNumber genRegArgNext(regNumber argReg)
 {
+    assert(isValidIntArgReg(argReg) || isValidFloatArgReg(argReg));
+
     regNumber result = REG_NA;
 
-    if (isValidFloatArgReg(argReg))
+#ifdef _TARGET_AMD64_
+#ifdef UNIX_AMD64_ABI
+    // Windows X64 ABI:
+    //     REG_EDI, REG_ESI, REG_ECX, REG_EDX, REG_R8, REG_R9
+    //
+    if (argReg == REG_ARG_1) // REG_ESI
     {
-        // We can iterate the floating point argument registers by using +1
-        result = REG_NEXT(argReg);
+        result = REG_ARG_2; // REG_ECX
     }
-    else
+    else if (argReg == REG_ARG_3) // REG_EDX
     {
-        assert(isValidIntArgReg(argReg));
-
-#ifdef _TARGET_AMD64_
-#ifdef UNIX_AMD64_ABI
-        // Windows X64 ABI:
-        //     REG_EDI, REG_ESI, REG_ECX, REG_EDX, REG_R8, REG_R9
-        //
-        if (argReg == REG_ARG_1) // REG_ESI
-        {
-            result = REG_ARG_2; // REG_ECX
-        }
-        else if (argReg == REG_ARG_3) // REG_EDX
-        {
-            result = REG_ARG_4; // REG_R8
-        }
+        result = REG_ARG_4; // REG_R8
+    }
+    else
 #else  // Windows ABI
-        // Windows X64 ABI:
-        //     REG_ECX, REG_EDX, REG_R8, REG_R9
-        //
-        if (argReg == REG_ARG_1) // REG_EDX
-        {
-            result = REG_ARG_2; // REG_R8
-        }
+    // Windows X64 ABI:
+    //     REG_ECX, REG_EDX, REG_R8, REG_R9
+    //
+    if (argReg == REG_ARG_1) // REG_EDX
+    {
+        result = REG_ARG_2; // REG_R8
+    }
+    else
```

Comment:
Please do not split the `else` statement from its block using `#ifdef`. It seems like this would be better off rewritten as a switch in any case:

```
switch (argReg)
{
#ifdef _TARGET_AMD64_
#ifdef UNIX_AMD64_ABI
    case REG_ARG_1:
        return REG_ARG_2;

    case REG_ARG_3:
        return REG_ARG_4;
#else
    case REG_ARG_1:
        return REG_ARG_2;
#endif

    default:
        return REG_NEXT(argReg);
}
```